web/src/echo.js

```diff
@@ -365,6 +365,14 @@ export function reify_message_id(local_id, server_id) {
     update_message_lists(opts);
     compose_notifications.reify_message_id(opts);
     recent_view_data.reify_message_id_if_available(opts);
+
+    if (message.type === "stream") {
+        stream_topic_history.reify_message_id_if_available({
+            ...opts,
+            stream_id: message.stream_id,
+            topic_name: message.topic,
+        });
+    }
```

Comment:
Is there a related bug if the message failed to send?

The alternative way we could approach fixing this is to not insert into `stream_topic_history` during local echo; that would be cleaner in some ways but might result in the left sidebar having an undesirable deferred-update experience.

Worth testing both approaches a bit, probably.

Reply:
Makes sense, I'll update it!

Reply:
updated!